### DISCUSSION TITLE
Sonoff Dual eratic behaviour on restart

### DISCUSSION THREAD

**Discussion by ghost** [arendst/Tasmota#15128] [Q&A] [answered]
Hi guys,

I'm struggling with the following problem.
I am using Tasmota on a Sonoff Dual R2 to control a shutter. This shutter is in front of a door so I've configured it in such a way that '0' means it is fully open. I've already had the shutter coming down twice while the door was open damaging the shutter before I did this.
Right now the following happens when I restart the device (or my NAS on which Home Assistant is running): 
If the shutter is fully up (so at position '0'), the shutter will move to the down position.
If the shutter is in any other position than fully up, the shutter moves to the fully up position.

I have another shutter running the same config, yet that one behaves as expected (after every reboot it moves to the fully up position).

I've tried resetting the device and check the logs for differences between both shutters but can't seem to find any. 
Could anyone please help me out (I'm more than willing to provide the necessary logs but I don't know which are relevant) so I can enjoy coming summer with my doors open without the fear of ruining my shutter?

Thanks a lot!

**Comment by barbudor** [comment]
Hi
> the following happens when I restart the device (or my NAS on which Home Assistant is running):

This points to a retained command topic which Tasmota sees either when Tasmota restart and subscrive to the cmnd topic or when the MQTT broker restart and resend the command

You can set `Weblog 3` and restart Tasmota. You should see an MQTT command telling Tasmota to go down.

**Reply by ghost** [reply]
Thanks for your quick reply. I've turned on weblog 3 and the following shows:

- 
- 21:56:01.096 MQT: Subscribe to cmnd/DVES_D86A32_fb/#
- 21:56:01.099 MQT: Subscribe to homeassistant/status
- 21:56:01.104 MQT: tele/tasmota_D86A32/INFO1 = {"Info1":{"Module":"Sonoff Dual R2","Version":"11.0.0(tasmota)","FallbackTopic":"cmnd/DVES_D86A32_fb/","GroupTopic":"cmnd/tasmotas/"}}
- 21:56:01.114 MQT: tele/tasmota_D86A32/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-D86A32-2610","IPAddress":"192.168.86.223"}}
- 21:56:01.126 MQT: tele/tasmota_D86A32/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":13}}
- 21:56:01.135 MQT: stat/tasmota_D86A32/RESULT = {"POWER1":"OFF"}
- 21:56:01.140 MQT: stat/tasmota_D86A32/POWER1 = OFF
- 21:56:01.144 MQT: stat/tasmota_D86A32/RESULT = {"POWER2":"OFF"}
- 21:56:01.151 MQT: stat/tasmota_D86A32/POWER2 = OFF
- 21:56:01.206 SRC: MQTT
- 21:56:01.208 CMD: Grp 0, Cmnd 'BACKLOG', Idx 1, Len 12, Data 'ShutterOpen1'
- 21:56:01.255 SRC: Backlog
- 21:56:01.257 CMD: Grp 0, Cmnd 'SHUTTEROPEN', Idx 1, Len 0, Data ''
- 21:56:01.263 MQT: stat/tasmota_D86A32/RESULT = {"Shutter1":{"Position":0,"Direction":0,"Target":0,"Tilt":0}}
- 21:56:01.267 MQT: stat/tasmota_D86A32/RESULT = {"ShutterOpen1":0}
- 21:56:01.271 SRC: MQTT
- 21:56:01.272 CMD: Grp 0, Cmnd 'SHUTTERPOSITION', Idx 1, Len 1, Data '0'
- 21:56:01.276 SRC: Shutter
- 21:56:01.286 MQT: tele/tasmota_D86A32/STATE = {"Time":"2022-03-13T21:56:01","Uptime":"0T00:00:06","UptimeSec":6,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":87,"MqttCount":1,"POWER1":"OFF","POWER2":"ON","Wifi":{"AP":1,"SSId":"Verkeerstoren","BSSId":"7C:2E:BD:7D:9C:94","Channel":6,"Mode":"11n","RSSI":100,"Signal":-44,"LinkCount":1,"Downtime":"0T00:00:03"}}
- 21:56:01.312 MQT: stat/tasmota_D86A32/RESULT = {"POWER2":"ON"}
- 21:56:01.316 MQT: stat/tasmota_D86A32/POWER2 = ON
- 21:56:01.321 MQT: stat/tasmota_D86A32/RESULT = {"ShutterPosition1":100}
- 21:56:01.327 SHT: Shtr1 Real 450000, Start 450000, Stop 0, Dir -1, Delay 0, Rtc 0.00 [s], Freq 0, PWM 0, Tilt 0
- 21:56:01.332 MQT: stat/tasmota_D86A32/RESULT = {"Shutter1":{"Position":0,"Direction":-1,"Target":100,"Tilt":0}}
- 21:56:01.390 SHT: Shtr1 Real 450000, Start 450000, Stop 0, Dir -1, Delay 0, Rtc 0.05 [s], Freq 0, PWM 0, Tilt 0
- 21:56:01.394 MQT: stat/tasmota_D86A32/RESULT = {"Shutter1":{"Position":0,"Direction":-1,"Target":100,"Tilt":0}}
- 21:56:02.389 SHT: Shtr1 Real 428060, Start 450000, Stop 0, Dir -1, Delay 0, Rtc 1.05 [s], Freq 0, PWM 0, Tilt 0
- 21:56:02.394 MQT: stat/tasmota_D86A32/RESULT = {"Shutter1":{"Position":5,"Direction":-1,"Target":100,"Tilt":0}}

I don't see the down command from the MQTT. Just a command where the device decides to turn on Power2 (which is down in my case).

**Comment by barbudor** [comment, marked as answer]
I see 2 suspect commands coming from MQTT
- ShutterOpen1
- ShutterPosition1 0
These are likely to be retained commans and probably the cause of your problems

You should clear them from your broker. See https://tasmota.github.io/docs/MQTT/#clearing-retained-messages

**Comment by ghost** [comment]
Thanks for your reply, I really appreciate your help.

I've followed your link and installed the Tasmota Device Manager and cleared the retained messages on the specified device.
The problem unfortunately still persists and the same commands from the MQTT are still there after rebooting.

EDIT: Somehow the device calls for 'ShutterPosition1: 100' after which the shutter starts to come down. I can't seem to find any reason why it does so.

**Reply by barbudor** [reply]
I've just tried the "clear retained" function in TDM and it doesn't seems to work. I open an issue there : https://github.com/jziolkowski/tdm/issues/188

Please try one of the other options. Personnaly I like MQTT Explorer
Or mosquitto command lines.

**Reply by sfromis** [reply]
Your ShutterOpen1 unambiguously comes in via MQTT. If not a retained message, it could also be some automation.

**Comment by ghost** [comment]
I used MQTT Explorer to clear the retained messages and it seems like that did the trick.

Thanks a lot!
